src/sentry/ratelimits/cardinality.py

```diff
@@ -228,13 +230,25 @@ def __init__(
             is a sampling rate, the lower it is, the less precise accounting
             will be.
         """
-        self.client = redis.redis_clusters.get(cluster)
+        self.is_redis_cluster, self.client, _ = redis.get_dynamic_cluster_from_options(
+            "", {"cluster": cluster}
+        )
+
+        self.helper: RedisHelper = (
+            RedisClusterHelper(self.client)
+            if self.is_redis_cluster
+            else RedisBlasterHelper(self.client)
+        )
+
         assert 0 < num_physical_shards <= num_shards
         self.num_shards = num_shards
         self.num_physical_shards = num_physical_shards
         self.metric_tags = metric_tags or {}
         super().__init__()
 
+    def validate(self) -> None:
+        redis.validate_dynamic_cluster(self.is_redis_cluster, self.client)
```

Comment:
Nobody seems to use this method.

Reply:
it's a method on `Service`, but since we don't use `LazyServiceWrapper` (which does call it) it's fair to say it doesn't matter whether we implement it. Still I think we should implement the interface properly.

Reply:
i actually went with removing this method to reduce the number of attributes that need to be stored.